nengo/synapses.py

```diff
@@ -466,8 +560,332 @@ class Alpha(LinearFilter):
+
+    Examples
+    --------
+    Plot frequency responses of bandpass filters centered around 2 Hz with varying
+    pass-band widths:
+
+    .. testcode::
+
+        freqs = np.logspace(-2, 2, 100)
+        for alpha in (0.1, 0.2, 0.5, 1, 2):
+            sys = nengo.synapses.Bandpass(2, alpha)
+            gain = np.abs(sys.evaluate(freqs))
+            plt.semilogx(
+                freqs,
+                20 * np.log10(gain),
+                label=r"$\alpha = %s$, $F_L = %0.3f$, $F_H = %0.3f$" % (
+                    alpha, sys.cutoff_low, sys.cutoff_high
+                )
+            )
+        plt.xlabel("Frequency (Hz)")
+        plt.ylabel("Gain (dB)")
+        plt.legend()
+    """
+
+    freq = NumberParam("freq", low=0, low_open=True)
+    alpha = NumberParam("alpha", low=0, low_open=True)
+
+    def __init__(self, freq, alpha=1, **kwargs):
+        self.freq = freq
+        self.alpha = alpha
+        w0 = freq * (2 * np.pi)
+        super().__init__(
+            ([alpha * w0, 0], [1, alpha * w0, w0 ** 2]), analog=True, **kwargs
+        )
+
+    @property
+    def cutoff_low(self):
+        """Low cutoff frequency in Hz; frequencies below this are attenuated."""
+        return (np.sqrt(1 + 0.25 * self.alpha ** 2) - 0.5 * self.alpha) * self.freq
+
+    @property
+    def cutoff_high(self):
+        """High cutoff frequency in Hz; frequencies above this are attenuated."""
+        return (np.sqrt(1 + 0.25 * self.alpha ** 2) + 0.5 * self.alpha) * self.freq
+
+
+class Highpass(LinearFilter):
+    r"""A highpass filter of given order.
+
+    The transfer function is given by:
+
+    .. math:: H(s) = \left( \frac{\tau s}{\tau s + 1} \right)^{order}
+
+    Equivalent to differentiating the input, scaling by :math:`\tau`,
+    lowpass filtering with time-constant :math:`\tau`, and finally repeating
+    this ``order`` times. The lowpass filter is required to make this causal.
+
+    The ``cutoff`` frequency is given by :math:`1 / (2 \pi \tau)`.
+
+    Parameters
+    ----------
+    tau : ``float``
+        Time-constant of the lowpass filter, and highpass gain.
+    order : ``integer``, optional
+        Dimension of the resulting linear system.
+
+    See Also
+    --------
+    :class:`.Lowpass`
+
+    Examples
+    --------
+    Evaluate the highpass in the frequency domain with a time-constant of 40 ms
+    (a cutoff of about 4 Hz), and order 2:
+
+    .. testcode::
+
+        sys = nengo.synapses.Highpass(0.04, order=2)
+        freqs = np.logspace(-2, 2, 100)
+        gain = np.abs(sys.evaluate(freqs))
+        plt.semilogx(freqs, 20 * np.log10(gain), label="cutoff = %0.3f" % (sys.cutoff))
+        plt.xlabel("Frequency (Hz)")
+        plt.ylabel("Gain (dB)")
+        plt.legend()
+    """
+
+    tau = NumberParam("tau", low=0)
+    order = IntParam("order", low=1)
+
+    def __init__(self, tau, order=1, **kwargs):
+        self.tau = tau
+        self.order = order
+        num = [tau] + [0] * order
```

Comment:
In nengolib this is `tau ** order` instead of `tau`. I think this is necessary to keep them unit gain. https://arvoelke.github.io/nengolib-docs/nengolib.synapses.Highpass.html 

![nengolib-synapses-Highpass-1](https://user-images.githubusercontent.com/5420057/91666033-c142e680-eac7-11ea-9e53-e2c42a921d4a.png)

Reply:
This change would also make it consistent with the equation in the class-level docstring (`.. math:: H(s) = \left( \frac{\tau s}{\tau s + 1} \right)^{order}`).

Reply:
Fixed. Also added a test that catches this (we were never testing `Highpass` with higher order before).